podwatcher/extension_test.go

```diff
@@ -0,0 +1,237 @@
+	stagingFullPatchExecutor   = `{"op":"add","path":"/spec/initContainers/0/command","value":["/bin/sh","-c","( /packs/executor \u0026\u0026 sleep 5 ) || sleep 5"]}`
+	stagingFullPatchDownloader = `{"op":"add","path":"/spec/initContainers/1/command","value":["/bin/sh","-c","( /packs/downloader \u0026\u0026 sleep 5 ) || sleep 5"]}`
+)
+
+var _ = Describe("Eirini extension", func() {
+	eirinixcat := eirinixcatalog.NewCatalog()
+	gracefulInjector := NewGracePeriodInjector(&GraceOptions{})
+	eiriniManager := eirinixcat.SimpleManager()
+	request := admission.Request{}
+	pod := &corev1.Pod{}
+
+	JustBeforeEach(func() {
+		gracefulInjector = NewGracePeriodInjector(&GraceOptions{})
+		eirinixcat = eirinixcatalog.NewCatalog()
+		eiriniManager = eirinixcat.SimpleManager()
+
+		raw, err := json.Marshal(pod)
+		if err != nil {
+			Expect(err).To(BeNil())
+		}
+
+		request = admission.Request{AdmissionRequest: admissionv1beta1.AdmissionRequest{Object: runtime.RawExtension{Raw: raw}}}
+	})
+
+	Describe("GracePeriod Injector", func() {
+		Context("when initializing", func() {
+			It("sets the default config", func() {
+				Expect(gracefulInjector.Options.FailGracePeriod).To(Equal("5"))
+				Expect(gracefulInjector.Options.SuccessGracePeriod).To(Equal("5"))
+				Expect(gracefulInjector.Options.StagingDownloaderEntrypoint).To(Equal("/packs/downloader"))
+				Expect(gracefulInjector.Options.StagingUploaderEntrypoint).To(Equal("/packs/uploader"))
+				Expect(gracefulInjector.Options.StagingExecutorEntrypoint).To(Equal("/packs/executor"))
+				Expect(gracefulInjector.Options.RuntimeEntrypoint).To(Equal("/lifecycle/launch"))
+			})
+
+			It("sets config", func() {
+				gracefulInjector = NewGracePeriodInjector(&GraceOptions{
+					FailGracePeriod:             "12",
+					SuccessGracePeriod:          "90",
+					StagingDownloaderEntrypoint: "foo",
+					StagingUploaderEntrypoint:   "bar",
+					StagingExecutorEntrypoint:   "baz",
+					RuntimeEntrypoint:           "42",
+				})
+
+				Expect(gracefulInjector.Options.FailGracePeriod).To(Equal("12"))
+				Expect(gracefulInjector.Options.SuccessGracePeriod).To(Equal("90"))
+				Expect(gracefulInjector.Options.StagingDownloaderEntrypoint).To(Equal("foo"))
+				Expect(gracefulInjector.Options.StagingUploaderEntrypoint).To(Equal("bar"))
+				Expect(gracefulInjector.Options.StagingExecutorEntrypoint).To(Equal("baz"))
+				Expect(gracefulInjector.Options.RuntimeEntrypoint).To(Equal("42"))
+			})
+		})
+	})
+
+	Describe("GracePeriod Injector", func() {
+		Context("when handling a Eirini runtime app", func() {
+			BeforeEach(func() {
+				pod = &corev1.Pod{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "opi",
+							},
+						},
+					},
+				}
+			})
+			It("Injects a grace period", func() {
+				Expect(decodePatches(gracefulInjector.Handle(context.TODO(), eiriniManager, pod, request))).To(Equal(addOpiPatch))
```

Comment:
Would it make sense to use [`MatchJSON`](http://onsi.github.io/gomega/#matchjsonjson-interface) instead? (For this and other comparisons)

Reply:
Well, yes and no, It would require to compose the full object, which seems a bit overkill here. Keep in mind that this extension shouldn't be around for too much (see https://github.com/cloudfoundry-incubator/kubecf/issues/1323 )